include/exadg/structure/spatial_discretization/operators/continuum_mechanics.h

```diff
@@ -30,40 +30,177 @@ namespace ExaDG
 template<int dim, typename Number>
 inline DEAL_II_ALWAYS_INLINE //
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  get_F(const dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> & H)
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>>
+  compute_H_plus_HT(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & H)
+{
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>> result;
+
+  for(unsigned int i = 0; i < dim; ++i)
+  {
+    for(unsigned int j = i; j < dim; ++j)
+    {
+      result[i][j] = H[i][j] + H[j][i];
+    }
+  }
+
+  // Debug output.
+  if constexpr(false)
+  {
+    if constexpr(std::is_same_v<Number, double>)
+    {
+      dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> check   = H + transpose(H);
+      dealii::VectorizedArray<Number>                         sum_err = 0.0;
+      for(unsigned int i = 0; i < dim; ++i)
+      {
+        for(unsigned int j = 0; j < dim; ++j)
+        {
+          sum_err += std::abs(check[i][j] - result[i][j]);
+        }
+      }
+      std::cout << "compute_H_plus_HT : sum_err = " << sum_err << "\n";
+      AssertThrow(sum_err.sum() < 1e-18,
+                  dealii::ExcMessage("Check in `compute_H_plus_HT()` failed."));
+    }
+  }
+
+  return result;
+}
+
+// Create a symmetric tensor from a tensor H^T times H.
+// Note that we have (H^T * H)^T = H^T * (H^T)^T = H^T * H.
+template<int dim, typename Number>
+inline DEAL_II_ALWAYS_INLINE //
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>>
+  compute_HT_times_H(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & H)
+{
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>> result;
+
+  for(unsigned int i = 0; i < dim; ++i)
+  {
+    for(unsigned int j = i; j < dim; ++j)
+    {
+      for(unsigned int k = 0; k < dim; ++k)
+      {
+        result[i][j] += /* HT[i][k] */ H[k][i] * H[k][j];
+      }
+    }
+  }
+
+  // Debug output.
+  if constexpr(false)
+  {
+    if constexpr(std::is_same_v<Number, double>)
+    {
+      dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> check   = transpose(H) * H;
+      dealii::VectorizedArray<Number>                         sum_err = 0.0;
+      for(unsigned int i = 0; i < dim; ++i)
+      {
+        for(unsigned int j = 0; j < dim; ++j)
+        {
+          sum_err += std::abs(check[i][j] - result[i][j]);
+        }
+      }
+      std::cout << "compute_HT_times_H : sum_err = " << sum_err << "\n";
+      AssertThrow(sum_err.sum() < 1e-18,
+                  dealii::ExcMessage("Check in `compute_HT_times_H()` failed."));
+    }
+  }
+
+  return result;
+}
+
+// Create a symmetric tensor from a tensor H times H^T.
+// Note that we have (H * H^T)^T = (H^T)^T * H^T = H * H^T.
+template<int dim, typename Number>
+inline DEAL_II_ALWAYS_INLINE //
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>>
+  compute_H_times_HT(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & H)
```

Comment:
note that we have now `compute_HT_times_H()` and `compute_H_times_HT` and the only thing that changes is the position of the transpose sign. The argument is *not* aussumed symmetric, so the results of the two functions are not the same. I did not want to create a single funtion with a boolean (template) parameter `transpose_first_else_second`, because I think this is then a bit confusing, especially in the code where this is then used. This function signature is a bit nicer to read, having
```
symmetric_tensor tmp = compute_H_times_HT<dim, Number>(src);
```
instead of
```
symmetric_tensor tmp = compute_H_times_H<dim, Number,false /* transpose_first_else_second */>(src);
```
.